implementations/micrometer-registry-wavefront/src/main/java/io/micrometer/wavefront/WavefrontMeterRegistry.java

```diff
@@ -303,8 +421,72 @@ public Builder httpClient(HttpSender httpClient) {
             return this;
         }
 
+        /**
+         * @param applicationTags   {@link ApplicationTags} containing metadata about your application that get reported to Wavefront as tags.
+         * @return The Wavefront registry builder with added {@link ApplicationTags}.
+         */
+        public Builder applicationTags(ApplicationTags applicationTags) {
+            this.applicationTags = applicationTags;
+            return this;
+        }
+
+        /**
+         * @return The Wavefront registry builder with aggregating of {@link WavefrontHistogram} by minute intervals enabled.
+         */
+        public Builder reportMinuteDistribution() {
+            histogramGranularities.add(HistogramGranularity.MINUTE);
+            return this;
+        }
+
+        /**
+         * @return The Wavefront registry builder with aggregating of {@link WavefrontHistogram} by hour intervals enabled.
+         */
+        public Builder reportHourDistribution() {
+            histogramGranularities.add(HistogramGranularity.HOUR);
+            return this;
+        }
+
+        /**
+         * @return The Wavefront registry builder with aggregating of {@link WavefrontHistogram} by day intervals enabled.
+         */
+        public Builder reportDayDistribution() {
+            histogramGranularities.add(HistogramGranularity.DAY);
+            return this;
+        }
+
+        @Nullable
+        private Tags globalTags() {
+            if (applicationTags == null) {
+                return null;
+            }
+            Tags globalTags = Tags.of(
```

Comment:
We cannot assume any sort of structure around common tags. What makes sense for an IaaS looks totally different on a CaaS, etc.

Reply:
`ApplicationTags` are a Wavefront-specific set of pre-defined tags.  They're optional here.

Reply:
This is not OK, leads to vendor-specific configuration of tags that should be common tags for portability.

Reply:
The furthest we've taken this in other implementations is to define which tag keys, if present, are mapped to some vendor-defined specific tag. See `hostTag` on `DatadogConfig`, for example. We still don't define the value of host in `DatadogConfig`, nor does the implementation assume that the tag key will be present on all meters necessarily.